citrination_client/views/data_view_builder.py

```diff
@@ -0,0 +1,152 @@
+
+        :param descriptor_key: Identifier for the descriptor
+        :param descriptor_values: An array of strings
+        :param role: Optionally specify a role (input, output, latentVariable, or ignored)
+        """
+        descriptor = dict(
+            category='Categorical',
+            descriptor_key=descriptor_key,
+            descriptor_values=descriptor_values
+        )
+        self.configuration['descriptors'].append(descriptor)
+        if role is not None:
+            self.set_role(descriptor_key, role)
+
+    def add_organic_descriptor(self, descriptor_key, role=None):
+        """
+        Add an organic valued descriptor to the configuration
+
+        :param descriptor_key: Identifier for the descriptor
+        :param role: Optionally specify a role (input, output, latentVariable, or ignored)
+        """
+        descriptor = dict(
+            category='Organic',
+            descriptor_key=descriptor_key
+        )
+        self.configuration['descriptors'].append(descriptor)
+        if role is not None:
+            self.set_role(descriptor_key, role)
+
+    def add_alloy_composition_descriptor(self, descriptor_key, balance_element, basis, threshold, role=None):
+        """
+        Add an alloy composition descriptor to the configuration
+
+        :param descriptor_key: Identifier for the descriptor
+        :param balance_element: The element the basis is applied to
+        :param basis: Percentage of the balance_element
+        :param threshold: Allowable deviation from basis
+        :param role: Optionally specify a role (input, output, latentVariable, or ignored)
+        """
+        descriptor = dict(
+            category='Alloy composition',
+            descriptor_key=descriptor_key,
+            balance_element=balance_element,
+            basis=basis,
+            threshold=threshold
+        )
+        self.configuration['descriptors'].append(descriptor)
+        if role is not None:
+            self.set_role(descriptor_key, role)
+
+    def add_inorganic_descriptor(self, descriptor_key, threshold, role=None):
+        """
+        Add an inorganic valued descriptor to the configuration
+
+        :param descriptor_key: Identifier for the descriptor
+        :param threshold: The closeness to an interpretable chemical formula that is accepted as valid
+        :param role: Optionally specify a role (input, output, latentVariable, or ignored)
+        """
+        descriptor = dict(
+            category='Inorganic',
+            descriptor_key=descriptor_key,
+            threshold=threshold
+        )
+        self.configuration['descriptors'].append(descriptor)
+        if role is not None:
+            self.set_role(descriptor_key, role)
+
+    def add_descriptor(self, descriptor, role=None):
+        """
+        Add a generic descriptor entry.
+
+        :param descriptor: Descriptor dict, see add_*_descriptor methods in this class for examples
```

Comment:
I feel pretty strongly that we should kill the specific `add_real_descriptor, ...` methods in favor of using citrination_client.models.columns.*. What's the argument against? Also, while we're here, we should add the group_by_key to a growing list and throw an error if you try to add two descriptors with the same name.

Reply:
That is easy enough on the pycc side, FE would need changes to deal with the modified schema going back up/down.  Down in the case of get which returns descriptors, which is what these are.  A hybrid option would be to have builder use them, but format them on request to match current endpoint spec.